src/handlers/payout/action.ts

```diff
@@ -441,68 +429,97 @@ export const handleIssueClosed = async (
-          reward.priceInEth
-        } ${incentivesCalculation.tokenSymbol.toUpperCase()} ]** ](${payoutUrl})\n`;
+    const detailsValue = reward.priceArray
+      .map((price, i) => {
+        const separateTitle = reward.type[i]?.split("-");
+        if (!separateTitle) return { title: "", subtitle: "", value: "" };
+        return { title: separateTitle[0], subtitle: separateTitle[1], value: price };
+      })
+      // remove title if it's the same as the first one
+      .map((item, i, arr) => {
+        if (i === 0) return item;
+        if (item.title === arr[0].title) return { ...item, title: "" };
+        return item;
+      });
+
+    const { reason, value } = await getWalletMultiplier(reward.user, incentivesCalculation.id?.toString());
+
+    // if reason is not "", then add multiplier to detailsValue and multiply the price
+    if (reason) {
+      detailsValue.push({ title: "Multiplier", subtitle: "Amount", value: value.toString() });
+      detailsValue.push({ title: "", subtitle: "Reason", value: reason });
+
+      const multiplier = new Decimal(value);
+      const price = new Decimal(reward.priceInEth);
+      // add multiplier to the price
+      reward.priceInEth = price.mul(multiplier);
+    }
+
+    const { payoutUrl, txData } = await generatePermit2Signature(reward.account, reward.priceInEth, reward.issueId, reward.userId?.toString());
+
+    const mergedType = reward.type.join(",");
+    const price = `${reward.priceInEth} ${incentivesCalculation.tokenSymbol.toUpperCase()}`;
+
+    switch (mergedType) {
+      case "Issue-Comments,Review-Reviewer":
+        comment = createDetailsTable(price, payoutUrl, reward.user, detailsValue);
+        break;
+      case "Issue-Comments":
+        comment = createDetailsTable(price, payoutUrl, reward.user, detailsValue);
+        break;
+      case "Review-Reviewer":
+        comment = createDetailsTable(price, payoutUrl, reward.user, detailsValue);
         break;
-      case "Conversation":
-        commentersComment = `${commentersComment}\n### [ **${reward.user}: [ CLAIM ${
-          reward.priceInEth
-        } ${incentivesCalculation.tokenSymbol.toUpperCase()} ]** ](${payoutUrl})\n`;
-        commentersReward[reward.user] = payoutUrl;
+      case "Issue-Creation,Issue-Assignee":
+        comment = createDetailsTable(price, payoutUrl, reward.user, detailsValue);
         break;
-      case "Reviewer":
-        pullRequestReviewerComment = `${pullRequestReviewerComment}\n### [ **${reward.user}: [ CLAIM ${
-          reward.priceInEth
-        } ${incentivesCalculation.tokenSymbol.toUpperCase()} ]** ](${payoutUrl})\n`;
-        prReviewersReward[reward.user] = payoutUrl;
+      case "Issue-Assignee":
+        comment = createDetailsTable(price, payoutUrl, reward.user, detailsValue);
+        break;
+      case "Issue-Creation":
+        comment = createDetailsTable(price, payoutUrl, reward.user, detailsValue);
         break;
       default:
         break;
     }
```

Comment:
I recall eight roles. There's six here?

Reply:
> Why did you merge two `case` with two roles? Shouldn't there be eight separate `case`s?

some roles are merged to get a new role.. for example, creator and assignee are two different roles but they are merged together when one user gets them both

It's also 6 because according to the original spec, creators should not get comment incentives and assignee should also not get comment incentives

Reply:
Perhaps I jumbled things up with another specification but I wanted it to be configurable in case partners did want to give comment incentives to assignees.

Reply:
> Perhaps I jumbled things up with another specification but I wanted it to be configurable in case partners did want to give comment incentives to assignees.

It was specified here: https://github.com/ubiquity/ubiquibot/issues/704#issue-1877717132

Reply:
Unfortunately it's really tedious for me to find the original spec / list from my phone but I want to clarify that there are eight recognized ways to contribute and we should stick with the same terms 

Prefix (which view) this part you got right

1. Issue
2. Review

Role:

1. Assignee
2. Issuer
3. Collaborator -> added to repo, has authority to leave official review states etc
4. Contributor -> everybody else

Reply:
I think you are getting it wrong, this is not a role.. these are the types of permit that are created

Assignee Permit, Issue Creator Permit, Reviewers Permit, Conversation Permits (For Commenters)

Reply:
@pavlovcik

Reply:
I think it works without the switch, refactored.. no need for switch type checks